app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -143,6 +145,23 @@ protected void onNewIntent(Intent unsafeIntent) {
             pendingUrl = intent.getDataString();
         }
 
+        if (Intent.ACTION_SEND.equals(intent.getAction())) {
```

Comment:
onNewIntent() is only called if the app is already running. Try killing Focus and then sharing to Focus. You will need to add a second entry point from onCreate(). See VIEW handling.